### ISSUE TITLE
Picking up another bullet casing when already having a same one in iventory cause bug

### ISSUE THREAD

**Issue by martinrhan** [CleverRaven/Cataclysm-DDA#40502]
# Describe the bug
I have one 7.62x39mm casing in my bag, then i picked up another one drop on ground after a gun shot. However I was told that I don't have enough capacity to take it. The number shown for the amount of 7.62x39mm casing that I want to pick up is -1.
using / item menu to pick it up cause game crashing.

# Steps To Reproduce
1,generate 2 such itmes.
2,drop one of them.
3,pick it up.

# Expected behavior
It should working as normal, no such bug.

# Versions and configuration
- OS: Windows
    - OS Version: 10.0 1903
- Game Version: 0.E-2083-g702b0b4 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Blaze Industries [blazeindustries],
    C.R.I.T Expansion Mod [crt_expansion],
    Magiclysm [magiclysm],
    Aftershock [aftershock],
    DinoMod [DinoMod],
    Mythical Martial Arts [MMA],
    No Rail Stations [No_Rail_Stations],
    Graphical Overmap [Graphical_Overmap],
    Stats Through Skills [StatsThroughSkills],
    SpeedyDex [speedydex],
    Stats Through Kills [stats_through_kills]
]

**Comment by KorGgenT** [comment]
can you attach a save with this bug?

**Comment by martinrhan** [comment]
[韦克罗斯.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/4619503/default.zip)
Actually this is not necessary because you can just make a new game to test

**Comment by martinrhan** [comment]
i found a way to avoid this, just drop the items in iventory to the ground to let them stack together then pick up. But I just found another BUG, I just butchered a zombie, and want to pick up its bone, but picked up bones are hold in hand and connot be inserted into my bag.

**Comment by martinrhan** [comment]
seem like any item that can have multiper number stacking togeter is buggy, I just tried pick up only one of them, fine, then pick up 10 of them, I received one extra.

**Comment by Lamandus** [comment]
![grafik](https://user-images.githubusercontent.com/33199510/82139459-2a0f9700-9828-11ea-8e68-1428201334bd.png)
I got the same error with other objects as well.

**Comment by Ichthysdekilt** [comment]
I'm also getting this same error.  Happens a lot with Eggs on a new experimental save (created on 10676).  Negative items can be merged into normal items (dropped together or put in the same container) to delete them and an equal number of normal items, but if you ignore them long enough they lose the "negative" listed with them - which doesn't turn them into normal items, mind.  Picking them up after this point causes a crash.  Safest way to keep playing with this bug is to burn your negative items or drop them to rot.

**Comment by peperov** [comment]
I did some digging. The problem emerges here:
https://github.com/CleverRaven/Cataclysm-DDA/blob/5ec73e76f557a337bf061ca9b93444975ba7fc9b/src/pickup.cpp#L265-L267
In commit 8513b65f85c6d0276d43b736df8519e4ef67bf2b this line was changed to what it is now. However as I was not involved in the container item changes, I don't think I'm able to fix this.

Just leaving this here as a lead/starting point for someboy more competent.
